pkg/controller/clusterinstallation/utils.go

```diff
@@ -112,14 +112,21 @@ func (r *ReconcileClusterInstallation) checkClusterInstallation(mattermost *matt
 	return status, nil
 }
 
-func (r *ReconcileClusterInstallation) checkSecret(secretName, namespace string) error {
+func (r *ReconcileClusterInstallation) checkSecret(secretName, keyName, namespace string) (bool, error) {
 	foundSecret := &corev1.Secret{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, foundSecret)
 	if err != nil {
-		return errors.Wrap(err, "Error getting secret")
+		return false, errors.Wrap(err, "Error getting secret")
 	}
 
-	return nil
+	for key := range foundSecret.Data {
+		if keyName == key {
+			return true, nil
+		}
+	}
+
+	msg := fmt.Sprintf("Missing required secret data. Want: %s", keyName)
+	return false, errors.Wrap(err, msg)
```

Comment:
If you are choosing to return an error here if the value isn't found, then there isn't much point to also returning a bool as well.

Couldn't we just infer that if there is no error, then the secret key was found?

Reply:
makes sense, done